File: Samples/General operations/CheckIfPasswordProtected/C#/CheckIfPasswordProtected.cs
using System;
using System.Linq;
using System.Text;

namespace BitMiracle.Docotic.Pdf.Samples
{
    public static class CheckIfPasswordProtected
    {
        public static void Main()
        {
            StringBuilder message = new StringBuilder();

            string[] documentsToCheck = { "encrypted.pdf", "jfif3.pdf", "public-key-encrypted.pdf" };
            foreach (string fileName in documentsToCheck)
            {
                PdfEncryptionInfo info = PdfDocument.GetEncryptionInfo(@"..\Sample Data\" + fileName);
                if (info == null)
                {
                    message.AppendFormat("{0} - is not encrypted\r\n", fileName);
                }
                else if (info is PdfStandardEncryptionInfo standardInfo)
                {
                    if (standardInfo.RequiresPasswordToOpen)
                        message.AppendFormat("{0} - is encrypted and requires password\r\n", fileName);
                    else
                        message.AppendFormat("{0} - is encrypted and doesn't require password\r\n", fileName);
                }
                else if (info is PdfPublicKeyEncryptionInfo publicKeyInfo)
                {
                    message.AppendFormat("{0} - is encrypted and requires one of the certificates: ", fileName);
                    string[] serialNumbers = publicKeyInfo.Recipients.Select(r => r.SerialNumber).ToArray();
                    message.Append(string.Join(", ", serialNumbers));
                }
            }

            Console.WriteLine(message.ToString());
        }
    }
}

File: Samples/General operations/CheckIfPasswordProtected/VB.NET/CheckIfPasswordProtected.vb
Imports System.Text

Imports BitMiracle.Docotic.Pdf

Namespace BitMiracle.Docotic.Pdf.Samples
    Public NotInheritable Class CheckIfPasswordProtected
        Public Shared Sub Main()
            Dim message As New StringBuilder()

            Dim documentsToCheck As String() = {"encrypted.pdf", "jfif3.pdf", "public-key-encrypted.pdf"}
            For Each fileName As String In documentsToCheck
                Dim info As PdfEncryptionInfo = PdfDocument.GetEncryptionInfo("..\Sample Data\" & fileName)
                If info Is Nothing Then
                    message.AppendFormat("{0} - is not encrypted" & vbCrLf, fileName)
                ElseIf TypeOf info Is PdfStandardEncryptionInfo Then
                    Dim standardInfo As PdfStandardEncryptionInfo = info
                    If standardInfo.RequiresPasswordToOpen Then
                        message.AppendFormat("{0} - is encrypted and requires password" & vbCrLf, fileName)
                    Else
                        message.AppendFormat("{0} - is encrypted and doesn't require password" & vbCrLf, fileName)
                    End If
                ElseIf TypeOf info Is PdfPublicKeyEncryptionInfo Then
                    Dim publicKeyInfo As PdfPublicKeyEncryptionInfo = info
                    message.AppendFormat("{0} - is encrypted and requires one of the certificates: ", fileName)
                    Dim serialNumbers As String() = publicKeyInfo.Recipients.Select(Function(r) r.SerialNumber).ToArray()
                    message.Append(String.Join(", ", serialNumbers))
                End If
            Next

            Console.WriteLine(message.ToString())
        End Sub
    End Class
End Namespace